controllers/model/modelversion_controller.go

```diff
@@ -0,0 +1,456 @@
+	}
+	exists := false
+	for _, ref := range modelVersion.OwnerReferences {
+		if ref.Kind == "Model" && ref.UID == model.UID {
+			// already exists
+			exists = true
+			break
+		}
+	}
+	if !exists {
+		modelVersion.OwnerReferences = append(modelVersion.OwnerReferences, metav1.OwnerReference{
+			APIVersion: model.APIVersion,
+			Kind:       model.Kind,
+			Name:       model.Name,
+			UID:        model.UID,
+		})
+	}
+}
+
+// createImgBuildPod creates a kaniko pod to build the image
+// 1. mount docker file into /workspace/dockerfile
+// 2. mount build source pvc into /workspace/build
+// 3. mount dockerconfig as /kaniko/.docker/config.json
+// The image build pod will create a container that includes artifacts under "/workspace/build", based on the dockerfile
+func createImgBuildPod(model *modelv1alpha1.ModelVersion, pvc *v1.PersistentVolumeClaim, imgBuildPodName string, newImage string) *v1.Pod {
+	podSpec := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imgBuildPodName,
+			Namespace: model.Namespace,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  imgBuildPodName,
+					Image: "gcr.io/kaniko-project/executor:latest",
+					Args: []string{
+						"--dockerfile=/workspace/dockerfile",
+						"--context=dir:///workspace/",
+						fmt.Sprintf("--destination=%s", newImage)},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	var volumeMounts = []v1.VolumeMount{
+		{
+			Name:      "kaniko-secret", // the docker secret for pushing images
+			MountPath: "/kaniko/.docker",
+		},
+		{
+			Name:      "build-source", // build-source references the pvc for the model
+			MountPath: "/workspace/build",
+		},
+		{
+			Name:      "dockerfile", // dockerfile is the default Dockerfile for building the image including the model
+			MountPath: "/workspace/",
+		},
+	}
+	var volumes = []v1.Volume{
+		{
+			Name: "kaniko-secret",
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: "regcred",
+					Items: []v1.KeyToPath{
+						{
+							Key:  ".dockerconfigjson",
+							Path: "config.json",
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "build-source",
+			VolumeSource: v1.VolumeSource{
+				PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+					ClaimName: pvc.Name,
+				},
+			},
+		},
+		{
+			Name: "dockerfile",
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: "dockerfile",
+					},
+				},
+			},
+		},
+	}
+
+	podSpec.Spec.Containers[0].VolumeMounts = volumeMounts
+	podSpec.Spec.Volumes = volumes
+	podSpec.OwnerReferences = append(podSpec.OwnerReferences, metav1.OwnerReference{
+		APIVersion: model.APIVersion,
+		Kind:       model.Kind,
+		Name:       model.Name,
+		UID:        model.UID,
+	})
+	return podSpec
+}
+
+func (r *ModelVersionReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	var predicates = predicate.Funcs{
```

Comment:
add modelversion to the ownerreference of pod and pvc.  modelversion controller watch the change of pod and pvc, after that there is no need to requeue after several seconds